src/parser/common.rs

```diff
@@ -170,6 +170,26 @@ pub fn parse_qubits(input: ParserInput) -> ParserResult<Vec<Qubit>> {
     many0(parse_qubit)(input)
 }
 
+/// Parse a variable qubit (i.e. a named qubit)
+pub fn parse_variable_qubit(input: ParserInput) -> ParserResult<String> {
+    match input.split_first() {
+        None => Err(nom::Err::Error(Error {
+            input,
+            error: ErrorKind::UnexpectedEOF("a variable qubit".to_owned()),
+        })),
+        Some((Token::Variable(name), remainder)) => Ok((remainder, name.clone())),
+        Some((Token::Identifier(name), remainder)) => Ok((remainder, name.clone())),
+        Some((other_token, _)) => {
+            expected_token!(input, other_token, stringify!($expected_variant).to_owned())
+        }
+    }
+}
+
+/// Parse zero or more variable qubits in sequence
+pub fn parse_variable_qubits(input: ParserInput) -> ParserResult<Vec<String>> {
```

Comment:
Adding this seems not worth the cost of having another `parse_X` function, to me. You only use it in one place, and typing `many0(parse_variable_qubit)(input)` directly doesn't seem like that much of a hardship.

Reply:
Ah, but I see we did the same with `parse_qubit`/`parse_qubits`. Carry on.

Reply:
Actually, yeah, I'll open a small PR removing these two.